src/core/restaurants/getAllRestaurantsSheets.tsx

```diff
@@ -0,0 +1,87 @@
+"use server";
+
+import { get } from "http";
+import { emitKeypressEvents } from "readline";
+import { RestaurantBooleanKeys, RestaurantInfos, restaurantBooleanKeys } from "./interface";
+
+const documentID = process.env.RESTAURANTS_SHEET_ID
+const valuesRange = process.env.SHEET_VALUE_RANGE
+const googleSheetsAPIKey = process.env.GOOGLE_SHEETS_API_KEY
+
+const sheetsColumnsToTechnicalName = {
+  Restaurant: 'name',
+  'Sur place': 'canEatIn',
+  'A emporter': 'canTakeAway',
+  Vege: 'vegetarianFriendly',
+  Viandard : 'meatLover',
+  "URL maps": "mapUrl",
+  "Moins de 10€": "lessThanTenEuros"
+} as const
+
+type GoogleSheetsExpectedColumnNames = keyof typeof sheetsColumnsToTechnicalName
+
+// Correct header type guard
+function isCorrectHeader(header: string[]): header is GoogleSheetsExpectedColumnNames[] {
+  const expectedHeader = Object.keys(sheetsColumnsToTechnicalName)
+  return header.every(col => expectedHeader.includes(col))
+}
+
+// Boolean key type guard
+function isBooleanKey(key: string): key is RestaurantBooleanKeys {
+  return restaurantBooleanKeys.includes(key)
+}
+
+
+const processBoolean = (value: string): boolean => {
```

Comment:
```suggestion
const processSheetsBooleanCell = (value: string): boolean => {
```

Pour apporter un peu plus de contexte sur ce que tu process :)

Reply:
Yes